massgov/mayflower
# DP-16115: Update feedback forms to allow toggling required asterisk on…

## Description
Update feedback forms to allow toggling required asterisk on and off.
Add a condition to add **required** attribute to radio button when it's required.
Sync the **required** value between **title** and **inputGroup**, so when **title.required** is _true_, **inputGroup.required** is also _true_.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-16115)
- [Github Mass PR](https://github.com/massgov/mass/pull/3471)

## Steps to Test

1. Visit https://mayflower.digital.mass.gov/b/patternlab/DP-6357--feedback-form-updates/?p=organisms-mass-feedback
2. Note that the form loads as expected.
The expectation here is that the form is _ready_ for required asterisks, but since none of the fields in the JSON are currently required, this form will function as it always has.

3. In the source, find the `legend` has `<span aria-hidden="true">*</span>` to prevent screen reader to announce the asterisk.

4. In the source, find the radio button `input` has **required** attribute to indicate it's required as shown below:
```<input name="foundIndicator" type="radio" value="Yes" id="field1" required="required">```

5. In /patternlab/styleguide/source/_patterns/03-organisms/feedback/mass-feedback.json, change L.5 `"requried": true"` to `"requried": false"`.  Find the asterisk doesn't show in the `legend` and `required` attribute is not added to the `input`:
```
<fieldset class="ma_feedback-fieldset feedback-load">
  <legend>
    Did you find what you were looking for on this webpage?
  </legend>

  <div class="ma__input-group ma__input-group--optional">
    <div class="ma__input-group__items ma__input-group__items--inline">
      <div class="ma__input-group__item">              
        <span class="ma__input-radio">
          <input name="foundIndicator" type="radio" value="Yes" id="field1">
          <label class="ma__input-radio__label" for="field1"><span>Yes</span></label>
        </span>
      </div>
      <div class="ma__input-group__item">            
        <span class="ma__input-radio">
          <input name="foundIndicator" type="radio" value="No" id="field2">
          <label class="ma__input-radio__label" for="field2"><span>No</span></label>
        </span>
      </div>
    </div>
  </div>
</fieldset>
```
